src/librados/librados_c.cc

```diff
@@ -41,13 +41,32 @@
 #define tracepoint(...)
 #endif
 
+#ifndef _WIN32
 #define LIBRADOS_C_API_BASE(fn)               \
   asm(".symver _" #fn "_base, " #fn "@")
 #define LIBRADOS_C_API_BASE_DEFAULT(fn)       \
   asm(".symver _" #fn ", " #fn "@@")
 #define LIBRADOS_C_API_DEFAULT(fn, ver)       \
   asm(".symver _" #fn ", " #fn "@@LIBRADOS_" #ver)
 
+#define LIBRADOS_C_API_BASE_F(fn) _ ## fn ## _base
+#define LIBRADOS_C_API_BASE_DEFAULT_F(fn) _ ## fn
+#define LIBRADOS_C_API_DEFAULT_F(fn, ver) _ ## fn
```

Comment:
if the version number is always dropped on the floor, why bother passing it?

Reply:
and why do we need to define macros for creating the function names?

Reply:
> if the version number is always dropped on the floor, why bother passing it?

Right, that's a leftover. Initially, I was thinking about appending the version to the function name.

> and why do we need to define macros for creating the function names?

On Linux, an underscore will be prepended to the function names and the right one will be picked through symbol versioning. This isnt't supported on Windows.